tests/test_scaleway_backend.py

```diff
@@ -0,0 +1,109 @@
+                "updated_at": "2023-01-01T00:00:00Z"
+            }
+        ]
+    }"""
+
+    def setUp(self):
+        super().setUp()
+        self.message = mail.EmailMultiAlternatives(
+            "Subject", "Text Body", "from@example.com", ["to@example.com"]
+        )
+
+
+@tag("scaleway")
+class ScalewayBackendStandardEmailTests(ScalewayBackendMockAPITestCase):
+    def test_send_mail(self):
+        mail.send_mail(
+            "Subject here",
+            "Here is the message.",
+            "from@example.com",
+            ["to@example.com"],
+            fail_silently=False,
+        )
+        self.assert_esp_called("/emails")
+        headers = self.get_api_call_headers()
+        self.assertEqual(headers["X-Auth-Token"], "test_api_key")
+        data = self.get_api_call_json()
+        self.assertEqual(data["subject"], "Subject here")
+        self.assertEqual(data["text"], "Here is the message.")
+        self.assertEqual(data["from"]["email"], "from@example.com")
+        self.assertEqual(data["to"][0]["email"], "to@example.com")
+
+    def test_html_message(self):
+        text_content = "This is an important message."
+        html_content = "<p>This is an <strong>important</strong> message.</p>"
+        email = mail.EmailMultiAlternatives(
+            "Subject", text_content, "from@example.com", ["to@example.com"]
+        )
+        email.attach_alternative(html_content, "text/html")
+        email.send()
+        data = self.get_api_call_json()
+        self.assertEqual(data["text"], text_content)
+        self.assertEqual(data["html"], html_content)
+
+    def test_attachments(self):
+        text_content = "* Item one\n* Item two\n* Item three"
+        self.message.attach(
+            filename="test.txt", content=text_content, mimetype="text/plain"
+        )
+        self.message.send()
+        data = self.get_api_call_json()
+        self.assertEqual(len(data["attachments"]), 1)
+        self.assertEqual(data["attachments"][0]["name"], "test.txt")
+
+    def test_api_failure(self):
+        self.set_mock_response(status_code=400)
+        with self.assertRaisesMessage(AnymailAPIError, "Scaleway API response 400"):
+            self.message.send()
+
+
+@tag("scaleway")
+class ScalewayBackendAnymailFeatureTests(ScalewayBackendMockAPITestCase):
+    def test_extra_headers(self):
+        self.message.extra_headers = {"X-Custom": "string", "X-Num": 123}
+        self.message.send()
+        data = self.get_api_call_json()
+        headers = {h["key"]: h["value"] for h in data["additional_headers"]}
+        self.assertEqual(headers["X-Custom"], "string")
+        self.assertEqual(headers["X-Num"], 123)
+
+    def test_reply_to(self):
+        self.message.reply_to = ["reply@example.com"]
+        self.message.send()
+        data = self.get_api_call_json()
+        headers = {h["key"]: h["value"] for h in data["additional_headers"]}
+        self.assertEqual(headers["Reply-To"], "reply@example.com")
+
```

Comment:
Since reply_to uses the same payload field as extra_headers, we should also have a test that uses both and makes sure additional_headers ends up with both of them.